src/schema/referencer.rs

```diff
@@ -38,6 +124,7 @@ pub(crate) struct ScalarTypeReferencers {
     pub(crate) object_field_arguments: IndexSet<ObjectFieldArgumentDefinitionPosition>,
     pub(crate) interface_fields: IndexSet<InterfaceFieldDefinitionPosition>,
     pub(crate) interface_field_arguments: IndexSet<InterfaceFieldArgumentDefinitionPosition>,
+    pub(crate) union_fields: IndexSet<UnionTypenameFieldDefinitionPosition>,
```

Comment:
isn't that union definition that they don't share any fields (as compared to interfaces)?

Reply:
See https://github.com/apollographql/federation-next/pull/80#discussion_r1383806697 . There's arguably a question of whether we really need to track this, but the JS code did track dependencies introduced by meta-fields (I haven't checked for usages of it yet though).